x-pack/functionbeat/provider/aws/sqs.go

```diff
@@ -51,3 +82,31 @@ func (s *SQS) createHandler(client core.Client) func(request events.SQSEvent) er
 func (s *SQS) Name() string {
 	return "sqs"
 }
+
+// Template returns the cloudformation template for configuring the service with the specified triggers.
+func (s *SQS) Template() *cloudformation.Template {
+	template := cloudformation.NewTemplate()
+
+	prefix := func(suffix string) string {
+		return "fnb" + s.config.Name + suffix
+	}
+
+	re := regexp.MustCompile("[:/-]")
```

Comment:
This is kind of a constant. Let's move the init out of this function and introduce a global:

```
var reNormalizeNames = regexp.MustCompile("[:/-]")
```

It is not nice to have var as global constant, but this patterns catches errors/panics whenever the beat is started, not only when the regex is compiled by taking said code path.

A little trick (lisp-style) if ouy want to hide the regex (still creates a function pointer, though):

var normalizeSQSName = func() func(string) string {
  re := regex.MustCompile()
  return func(c string) string {
    return re.ReplaceAllString(c, "")
  }
}()

Reply:
I have added the above method in common/string.go and tests for it.